packages/back-end/src/app.ts

```diff
@@ -145,6 +147,27 @@ if (ENVIRONMENT !== "production") {
   app.set("json spaces", 2);
 }
 
+if (stringToBoolean(process.env.PYTHON_SERVER_MODE)) {
+  // increase max payload json size to 50mb as a single query can return up to 3000 rows
+  // and we pass the results of all queries at once into python
+  app.use(
+    bodyParser.json({
+      limit: process.env.PYTHON_SERVER_INPUT_SIZE_LIMIT || "50mb",
+    })
+  );
+  app.use(compression());
+  app.use(httpLogger);
+  app.post("/stats", async (req, res) => {
+    try {
+      const results = await runStatsEngine(req.body);
+      res.status(200).json({ results });
+    } catch (error) {
+      logger.error("Error running stats engine:", error);
```

Comment:
Pino actually expects the error (if existing) as the first argument -- I have a separate PR fixing a lot of cases where we are not doing that, but we should update it here

```suggestion
      logger.error(error, "Error running stats engine");
```